src/sync/SplitUpdateWorker/index.js

```diff
@@ -57,8 +62,12 @@ export default class SplitUpdateWorker {
    * @param {string} defaultTreatment default treatment value
    */
   killSplit(changeNumber, splitName, defaultTreatment) {
-    this.splitStorage.killLocally(splitName, defaultTreatment, changeNumber);
-    this.put(changeNumber);
+    this.splitStorage.killLocally(splitName, defaultTreatment, changeNumber).then((updated) => {
+      if (updated) {
```

Comment:
A favor, please add a big fat note here that, whenever we support an async storage that the SDK itself writes to, we need retries or something in the event of a kill notification.